File: Oats/Quake3.lua
function Q_rsqrt(number)
  x2 = number * 0.5
  y = number
  
  i = string.pack(j, y)
  i = 0x5f3759df - (i >> 1)
  y = string.unpack(i)

  y = y * (1.5 - (x2 * y * y))
  y = y * (1.5 - (x2 * y * y))
  
  return y
end

print(Q_rsqrt(123.456))

File: Oats/Quake3.pl
use strict;
use warnings;

sub q_rsqrt {
    my $number = shift;
    my $three_halfs = 1.5;
    my $x2 = $number * 0.5;
    my $y = $number;
    my $i = unpack('l', pack('f', $y));
    $i = 0x5f3759df - ($i >> 1); 
    $y = unpack('f', pack('l', $i));
    $y = $y * ($three_halfs - ($x2 * $y * $y)); 

    return $y;
}

print q_rsqrt(2.43);
